### PR TITLE
gr_modtool: fix for yaml generator

### PR DESCRIPTION
**Pull request by aru31**
Fixes for issues in yaml generator in gr_modtool as mentioned in PR #2418.

### CODE REVIEW

**Comment by aru31** [comment]
@drmpeg, please review my PR.
Thanks. :)

**Comment by drmpeg** [comment]
Looks good. The other issues I mentioned are actually a corner case. They occur if the block name starts with the module name. You can test that case with:

gr_modtool newmod gizmo

gr_modtool add gizmo_bb
sync
cpp
et cetera

gr_modtool makeyaml
gizmo_bb

Can that be addressed?

**Comment by aru31** [comment]
@drmpeg, corner cases are now handled. :)

**Comment by drmpeg** [comment]
Works great! Thank you for your patience.

File: gr-utils/python/modtool/core/makeyaml.py [hunk 1]
```diff
@@ -147,13 +147,16 @@ def _type_translate(p_type, default_v=None):
                 return 'raw'
         def _get_blockdata(fname_cc):
             """ Return the block name and the header file name from the .cc file name """
+            flag = False
```
Comment (noc0lour) [inline gr-utils/python/modtool/core/makeyaml.py]: Please give the variable a meaningful name which explains its existence at first glance.

File: gr-utils/python/modtool/core/makeyaml.py [hunk 2]
```diff
@@ -147,13 +147,16 @@ def _type_translate(p_type, default_v=None):
...
             blockname = os.path.splitext(os.path.basename(fname_cc.replace('_impl.', '.')))[0]
             fname_h = (blockname + '.h').replace('_impl.', '.')
+            if self.info['modname']+'_' in blockname:
```
Comment (noc0lour) [inline gr-utils/python/modtool/core/makeyaml.py]: Would be testing with `startswith()` here more meaningful?

File: gr-utils/python/modtool/core/makeyaml.py (hunk 1 above)
Comment (aru31) [reply gr-utils/python/modtool/core/makeyaml.py]: Sure!

File: gr-utils/python/modtool/core/makeyaml.py (hunk 2 above)
Comment (aru31) [reply gr-utils/python/modtool/core/makeyaml.py]: Yes! Thank you.

**Comment by aru31** [comment]
@noc0lour, I have made the changes you requested. :)

File: gr-utils/python/modtool/core/makeyaml.py [hunk 3]
```diff
@@ -147,13 +147,16 @@ def _type_translate(p_type, default_v=None):
                 return 'raw'
         def _get_blockdata(fname_cc):
             """ Return the block name and the header file name from the .cc file name """
+            blockname_boolean = False
             blockname = os.path.splitext(os.path.basename(fname_cc.replace('_impl.', '.')))[0]
             fname_h = (blockname + '.h').replace('_impl.', '.')
+            if blockname.startswith(self.info['modname']+'_'):
```
Comment (noc0lour) [inline gr-utils/python/modtool/core/makeyaml.py]: Yesterday I thought about a simplification of the syntax. You can just run `contains_modulename = blockname.startswith(....`

Comment (noc0lour) [reply gr-utils/python/modtool/core/makeyaml.py]: Without any if-branching

Comment (aru31) [reply gr-utils/python/modtool/core/makeyaml.py]: Thanks. I have made the changes.
